js/main.js

```diff
@@ -284,3 +457,213 @@ function getJoinedDate(timestamp) {
+      if (flag1 === true && pgn[i] === '.') {
+        flag2 = true;
+      }
+    } else {
+      if (pgn[i] === '.') {
+        flag2 = true;
+      }
+    }
+
+    // penultimate character is always a number 0, 1, or 2
+    if (i === pgn.length - 2) {
+      if (pgn[i] === '0') {
+        if (username === white) {
+          result = 'Win';
+          colorStr = 'text-green';
+          bgColorStr = 'bg-win';
+          data.win++;
+        } else {
+          result = 'Loss';
+          colorStr = 'text-red';
+          bgColorStr = 'bg-loss';
+          data.loss++;
+        }
+      } else if (pgn[i] === '1') {
+        if (username === black) {
+          result = 'Win';
+          colorStr = 'text-green';
+          bgColorStr = 'bg-win';
+          data.win++;
+        } else {
+          result = 'Loss';
+          colorStr = 'text-red';
+          bgColorStr = 'bg-loss';
+          data.loss++;
+        }
+      } else if (pgn[i] === '2') {
+        result = 'Draw';
+        colorStr = 'text-gray';
+        bgColorStr = 'bg-draw';
+        data.draw++;
+      }
+    }
+    i--;
+  }
+
+  return { resultStr: result, moves: moveCount.join(''), color: colorStr, bgColor: bgColorStr };
+}
+
+// archive endpoint last part example: ..games/2021/07"
+function getMonthAndYear(endpointStr) {
+  const year = [];
+  const month = [];
+  let monthStr = '';
+  let monthIndex = 0;
+  let condition = false;
+  let count = 0;
+  let i = endpointStr.length - 1;
+
+  while (count < 8) {
+    if (endpointStr[i] === '/') {
+      condition = true;
+    }
+    if (endpointStr[i] !== '/') {
+      if (!condition) {
+        month.unshift(endpointStr[i]);
+      } else {
+        year.unshift(endpointStr[i]);
+      }
+    }
+    count++;
+    i--;
+  }
+  monthStr = month.join('');
+
+  if (month[0] === '0') {
+    monthIndex = Number(monthStr.slice(1)) - 1;
+  } else {
+    monthIndex = Number(monthStr) - 1;
+  }
+
+  return `${months[monthIndex]} ${year.join('')}`;
+}
+
+function updateWPCTElements() {
+  $winPCT.textContent = `${data.getWPCT()}%`;
+
+  for (let i = 0; i < $wdl.length; i++) {
+    switch (i) {
+      case 0:
+        $wdl[i].textContent = data.win;
+        break;
+      case 1:
+        $wdl[i].textContent = '|';
+        break;
+      case 2:
+        $wdl[i].textContent = data.draw;
+        break;
+      case 3:
+        $wdl[i].textContent = '|';
+        break;
+      case 4:
+        $wdl[i].textContent = data.loss;
+        break;
+    }
+  }
+  data.resetWDL();
+}
+
+function clearWPCTElement() {
+  $winPCT.textContent = '';
+  for (let i = 0; i < $wdl.length; i++) {
+    $wdl[i].textContent = '';
+  }
+}
+
+function clearTableElements() {
+  while ($statsTable.firstChild) {
+    $statsTable.removeChild($statsTable.firstChild);
+  }
+  while ($clubsTable.firstChild) {
+    $clubsTable.removeChild($clubsTable.firstChild);
+  }
+  while ($matchList.firstChild) {
+    $matchList.removeChild($matchList.firstChild);
+  }
```

Comment:
Since you also created functions that do these same things, you can just call those functions here rather than rewrite it. 

```
function clearTableElements() {
   clearStats();
   clearClubs();
   clearMatchList();
}
```